resources/assets/coffee/_classes/store-supporter-tag.coffee

```diff
@@ -0,0 +1,209 @@
+#    as published by the Free Software Foundation.
+#
+#    osu!web is distributed WITHOUT ANY WARRANTY; without even the implied
+#    warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#    See the GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with osu!web.  If not, see <http://www.gnu.org/licenses/>.
+###
+
+class @StoreSupporterTag
+  RESOLUTION: 8
+  MIN_VALUE: 4
+  MAX_VALUE: 52
+
+  CUTOFFS: {
+    2: 8,
+    4: 12,
+    6: 16,
+    8: 20,
+    9: 22,
+    10: 24,
+    12: 28,
+    18: 39,
+    24: 52,
+  }
+
+  @Price = {
+    price: 0
+    duration: 0
+    pricePerMonth: ->
+      (@price / @duration).toFixed(2)
+    discount: ->
+      raw = if @duration >= 12 then 46 else ((1 - (@price / @duration) / 4) * 100)
+      Math.max(0, Math.round(raw, 0))
+    durationInYears: ->
+      { years: Math.floor(@duration / 12), months: Math.floor(@duration % 12) }
+    durationText: ->
+      obj = @durationInYears()
+      yearsText = switch obj.years
+                  when 0
+                    ''
+                  when 1
+                    "#{obj.years} year"
+                  else
+                    "#{obj.years} years"
+
+      monthsText = switch obj.months
+                  when 0
+                    ''
+                  when 1
+                    "#{obj.months} month"
+                  else
+                    "#{obj.months} months"
+
+      _.compact([yearsText, monthsText]).join(', ')
+  }
+
+  constructor: ->
+    $(document).on 'turbolinks:load', =>
+      @initialize(document.getElementById('js-store-supporter-tag'))
+
+  initialize: (rootElement) =>
+    return unless rootElement
+    @el = rootElement
+    @searching = false
+    @searchData = null
+    @currentUser = {
+      username: @el.dataset.username
+      avatar_url: @el.dataset.avatarUrl
+    }
+    # Everything should be scoped under the root @el
+    @priceElement = @el.querySelector('.js-price')
+    @durationElement = @el.querySelector('.js-duration')
+    @pricePerMonthElement = @el.querySelector('.js-price-per-month')
+    @discountElement = @el.querySelector('.js-discount')
+    @slider = @el.querySelector('.js-slider')
+    @sliderPresets = @el.querySelectorAll('.js-slider-preset')
+    @usernameInput = @el.querySelector('#username.form-control')
+
+    @debouncedGetUser = _.debounce @getUser, 300
+    @initializeSlider()
+    @initializeSliderPresets()
+    @initializeUsernameInput()
+    @
+
+  initializeSlider: =>
+    slider = $(@slider).slider {
+      range: 'min',
+      value: @sliderValue(@MIN_VALUE),
+      min: @sliderValue(@MIN_VALUE),
+      max: @sliderValue(@MAX_VALUE),
+      step: 1,
+      animate: true,
+      slide: (event, ui) =>
+        @onSliderValueChanged event, ui
+      change: (event, ui) =>
+        @onSliderValueChanged event, ui
+    }
+    @updatePrice(@calculate(@sliderValue(@MIN_VALUE)))
+    slider
```

Comment:
it makes more sense for the function interface to return the slider instead of the output of @updatePrice, imo